### PR TITLE
Simplify startup code in library_browser.js. NFC

### PR DESCRIPTION
**Pull request by sbc100**
Also apply this simplification in a few other places in the JS library.

### CODE REVIEW

File: src/library_idbfs.js [hunk 1]
```diff
@@ -220,15 +220,15 @@ addToLibrary({
...
+      req.onsuccess = callback;
       req.onerror = (e) => {
         callback(e.target.error);
         e.preventDefault();
       };
     },
     removeRemoteEntry: (store, path, callback) => {
       var req = store.delete(path);
-      req.onsuccess = () => { callback(null); };
+      req.onsuccess = callback;
```
Comment (RReverser) [inline src/library_idbfs.js:231]: This doesn't seem right? In other places `onsuccess` gets an `event` argument, which in the previous version was simply ignored and callback was invoked with `null`, whereas now the event will be propagated to the callback which will think it's an error argument instead.

Risks like this is one of the reasons I'm a bit hesitant about changes like this. Brings to my mind this Jake Archibald's post: https://jakearchibald.com/2021/function-callback-risks/

The only safe simplification here seems to be `() => callback()` as that's what some other places are already doing.

Comment (sbc100) [reply src/library_idbfs.js:231]: Ah yes, good point.

File: src/IDBStore.js [hunk 2]
```diff
@@ -73,41 +73,39 @@ var IDBStore = {
...
-        callback(error);
-      };
+      req.onerror = callback;
     });
   },
   setFile(dbName, id, data, callback) {
     IDBStore.getStore(dbName, 'readwrite', (err, store) => {
       if (err) return callback(err);
       var req = store.put(data, id);
       req.onsuccess = (event) => callback();
```
Comment (RReverser) [inline src/IDBStore.js:83]: As mentioned in previous comment, you can probably remove `event` here unless you want to keep it for clarity.

File: src/library_idbfs.js [hunk 3]
```diff
@@ -220,15 +220,15 @@ addToLibrary({
         callback(e);
         return;
       }
-      req.onsuccess = () => { callback(null); };
+      req.onsuccess = (event) => { callback(); };
```
Comment (RReverser) [inline src/library_idbfs.js:223]: Can probably remove braces here.

**Review by RReverser — APPROVED** [review]
LGTM now % nits.

**Pull request merged**